gbpservice/neutron/nsf/core/threadpool.py

```diff
@@ -0,0 +1,80 @@
+import time
+import eventlet
+eventlet.monkey_patch()
+from eventlet import event
+from eventlet import greenpool
+from eventlet import greenthread
+import os
+import sys
+import threading
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+def _thread_done(gt, *args, **kwargs):
+    kwargs['pool'].thread_done(kwargs['thread'])
+
+
+class Thread(object):
+
+    def __init__(self, thread, pool):
+        self.thread = thread
+        self.thread.link(_thread_done, pool=pool, thread=self)
```

Comment:
It calls the _thread_done() method whenever the thread stops.. in case we have to handle anything.. we are removing the thread object from our cache.